code/script.js

```diff
@@ -218,108 +226,238 @@ const generateBoard = () => {
+  loadingGifandBoard();
+};
 
-  // Invoke a function to redraw the board with the remaining people.
-}
+const loadingGifandBoard = () => {
+  //empty the board
+  board.innerHTML = '';
+  //put a loading gif to make it looks like the computer is progressing
+  board.innerHTML += `        
+  <img src="/gif/beanEater.gif" class="loading-gif" alt="loading gif"/>
+`;
+  //after 1.5s redraw the board
+  setTimeout(() => generateBoard(), 1500);
+};
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+//the player have to confirm that they want to make a guess
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  //store the interaction from the player in variable, confirmGuess
+  let confirmGuess = confirm(
+    `Do you really want to guess on ${personToConfirm}?`
+  );
+  // If the player wants to guess, invoke the checkMyGuess function
+  if (confirmGuess === true) {
+    checkMyGuess(personToConfirm);
+  }
+};
 
-// If you confirm, this function is invoked
+// If you confirm and pressed OK, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+  // Check if the personToCheck is the same as the secret person's name
+  // Set a Message to show in the win or lose section accordingly
+  if (personToCheck === secret.name) {
+    winOrLoseText.innerHTML = `
+    You win!
+    `;
+    winAudio.play(); //play correct answer
+    winOrLoseText.innerHTML += `
+    <h1>You have guessed: ${countGuesses} time(s)!</h1>
+    `;
+  } else {
+    winOrLoseText.innerHTML = `
+    You lose!
+    `;
+    loseAudio.play(); //play wrong answer
+    winOrLoseText.innerHTML += `
+    <h1>You have guessed: ${countGuesses} time(s)!</h1>
+    `;
+  }
+  winOrLose.style.display = 'flex'; // Show the win or lose section
+  board.style.display = 'none'; // Hide the game board
+};
 
 // Invokes the start function when website is loaded
-start()
+start();
 
 // All the event listeners
-restartButton.addEventListener('click', start)
+restartButton.addEventListener('click', () => {
+  countGuesses = 0; //reset the counter for how many guesses
+  questions.value = ''; //empty select value when clicking the Restart button
+  loadingGifandBoard();
+  start();
+});
+
+//when selecting an option from the drop down list, it will invoke the selectQuestion function
+questions.addEventListener('change', selectQuestion);
+
+//Find out button, it will invoke the checkQuestion function
+findOutButton.addEventListener('click', () => {
+  checkQuestion();
+  countGuesses++; //counter for how many guesses, count each valid click on the Find out button
```

Comment:
The counter is really nice! Will try that out. Didn´t have time for it, and it seemed too hard. 😅

Reply:
I wanted to try the timer too but grown tired of the game... 😅  so I'm satisfied with the counter 🥰